From 0f6f4a8f1a9b826d36589987344ddc24e47c7f8d Mon Sep 17 00:00:00 2001
From: "Luke A. Guest" <laguest@archeia.com>
Date: Fri, 3 Feb 2017 15:56:58 +0000
Subject: [PATCH 08/14] Change second parameter passed to
 drm_vma_node_verify_access() from filp to filp->private_data, for kernel
 v4.10.

---
 amd/amdgpu/amdgpu_ttm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/amd/amdgpu/amdgpu_ttm.c b/amd/amdgpu/amdgpu_ttm.c
index 447529d..fa8998c 100644
--- a/amd/amdgpu/amdgpu_ttm.c
+++ b/amd/amdgpu/amdgpu_ttm.c
@@ -255,7 +255,11 @@ static int amdgpu_verify_access(struct ttm_buffer_object *bo, struct file *filp)
 
 	if (amdgpu_ttm_tt_get_usermm(bo->ttm))
 		return -EPERM;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0)
+	return drm_vma_node_verify_access(&abo->gem_base.vma_node, filp->private_data);
+#else
 	return drm_vma_node_verify_access(&abo->gem_base.vma_node, filp);
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0) */
 }
 
 static void amdgpu_move_null(struct ttm_buffer_object *bo,
-- 
2.7.3

